bcipy/helpers/tests/test_stimuli.py

```diff
@@ -221,6 +229,263 @@ def test_alphabetical_inquiry_gen(self):
+        """Test generation of inquiries with distributed target positions and no nontarget inquiries."""
+        alp = [
+            'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+            'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+            '<', '_'
+        ]
+        stim_number = 10
+        stim_length = 10
+        nontarget_inquiries = 0
+        inquiries, inq_timings, inq_colors = calibration_inquiry_generator(
+            alp,
+            timing=[0.5, 1, 0.2],
+            color=['green', 'red', 'white'],
+            stim_number=stim_number,
+            stim_length=stim_length,
+            stim_order=StimuliOrder.RANDOM,
+            target_positions=TargetPositions.DISTRIBUTED,
+            nontarget_inquiries=nontarget_inquiries,
+            is_txt=True)
+
+        self.assertEqual(
+            len(inquiries), stim_number,
+            'Should have produced the correct number of inquiries')
+        self.assertEqual(len(inq_timings), stim_number)
+        self.assertEqual(len(inq_colors), stim_number)
+
+        inq_strings = []
+        for inq in inquiries:
+            self.assertEqual(
+                len(inq), stim_length + 2,
+                ('inquiry should include the correct number of choices as ',
+                 'well as the target and cross.'))
+            choices = inq[2:]
+            self.assertEqual(stim_length, len(set(choices)),
+                             'All choices should be unique')
+
+            # create a string of the options
+            inq_strings.append(''.join(choices))
+
+        self.assertEqual(
+            len(inquiries), len(set(inq_strings)),
+            'All inquiries should be different')
+
+    def test_distributed_target_positions(self):
+        """Test generation of distributed target positions with nontarget inquiries."""
+
+        stim_number = 11
+        stim_length = 10
+        nontarget_inquiries = 10
+
+        target_inquiries = stim_number - (stim_number * (nontarget_inquiries / 100))
+        num_target_inquiries = (int)(target_inquiries / stim_length)
+
+        targets = distributed_target_positions(stim_number=stim_number,
+                                               stim_length=stim_length,
+                                               nontarget_inquiries=nontarget_inquiries)
+
+        self.assertEqual(
+            len(targets), stim_number,
+            'Should have produced the correct number of targets for inquiries.')
+
+        # count how many times each target position is used
+        target_counts = (np.zeros(stim_length + 1)).astype(int)
+        for pos in targets:
+            target_counts[pos] = (target_counts[pos] + 1)
+
+        # make sure position counts are equally distributed, including non-target
+        for i in target_counts:
+            self.assertTrue(i >= num_target_inquiries)
+            self.assertTrue(i <= (num_target_inquiries + 1))
```

Comment:
You can use chained comparisons on all of these.

https://docs.python.org/3/reference/expressions.html#comparisons